network/p2p/connGater.go

```diff
@@ -44,10 +60,13 @@ func (c *ConnGater) InterceptAddrDial(_ peer.ID, ma multiaddr.Multiaddr) bool {
 
 // InterceptAccept is not used. Currently, allowlisting is only implemented by Peer IDs and not multi-addresses
 func (c *ConnGater) InterceptAccept(cm network.ConnMultiaddrs) bool {
+	if c.onInterceptAccept != nil {
+		return c.onInterceptAccept(cm.RemoteMultiaddr())
```

Comment:
does `cm.RemoteMultiaddr()` libp2p use ephemeral ports when connecting? Have you confirmed this will match what's in the identity list?